app/scripts/controllers/transactions/lib/util.js

```diff
@@ -97,3 +98,6 @@ function getFinalStates () {
   ]
 }
 
+function containsFinalStates (txs) {
+  return !!txs.find((tx) => getFinalStates().includes(tx.status))
```

Comment:
We can move the call to `getFinalStates` outside the find to avoid creating a new array of states for each tx:

```js
const finalStates = getFinalStates()
return !!txs.find((tx) => finalStates.includes(tx.status))
```

Reply:
this is future proofing getFinalStates could be added to by plugin other creative things we do where new types of tx-flows are required and thus generating new final states

however if your not a fan i can change that

Reply:
Moving the function call outside of the `find` call shouldn't prohibit the `getFinalStates` function coming from somewhere else, but this is fine for now